jade-tui

```diff
@@ -0,0 +1,217 @@
+if [[ $CONTINUE != "true" ]]; then
+	echo "Exiting. Have a good day!"
+	exit
+fi
+
+# Select timezone, keymap, locale
+timezone=$(timedatectl list-timezones | gum filter --placeholder "select a timezone")
+keymap=$(localectl list-keymaps | gum filter --placeholder "select a keymap")
+locale=$(cat /usr/share/jade-tui/locales | gum filter --placeholder "select a locale")
+
+# Username
+clear
+gum style --border normal --margin "1" --padding "1 2" --border-foreground 212 "Please enter your username"
+username=$(gum input --placeholder "Please enter your username")
+
+# User password
+matches="false"
+passwrong="false"
+while [[ "$matches" == "false" ]]; do
+	clear
+	if [[ "$passwrong" == "true" ]]; then
+		gum style --border normal --margin "1" --padding "1 2" --border-foreground 212 "Passwords did not match, please type the password again"
+	else
+		gum style --border normal --margin "1" --padding "1 2" --border-foreground 212 "Now enter your password"
+	fi
+	password=$(gum input --password --placeholder "Please enter a password")
+	clear
+	gum style --border normal --margin "1" --padding "1 2" --border-foreground 212 "Verify your password"
+	password_verif=$(gum input --password --placeholder "Type your password again")
+	if [[ "$password" == "$password_verif" ]]; then
+		matches="true"
+	else
+		passwrong="true"
+	fi
+done
+crypt_password=$(openssl passwd -crypt $password)
+
+# Root password
+clear
+different_root_password=true
+gum confirm "$(gum style --border normal --margin '1' --padding '1 2' --border-foreground 212 'Use same password for root?')" && different_root_password=false
+if [[ $different_root_password != "true" ]]; then
+	root_password=$password # set root password same as user password
+else
+	root_matches="false"
+	root_passwrong="false"
+	while [[ "$root_matches" == "false" ]]; do
+		clear
+		if [[ "$root_passwrong" == "true" ]]; then
+			gum style --border normal --margin "1" --padding "1 2" --border-foreground 212 "Passwords did not match, please type the root password again"
+		else
+			gum style --border normal --margin "1" --padding "1 2" --border-foreground 212 "Now enter your root password"
+		fi
+		root_password=$(gum input --password --placeholder "Please enter a root password")
+		clear
+		gum style --border normal --margin "1" --padding "1 2" --border-foreground 212 "Verify your root password"
+		root_password_verif=$(gum input --password --placeholder "Type your root password again")
+		if [[ "$root_password" == "$root_password_verif" ]]; then
+			root_matches="true"
+		else
+			root_passwrong="true"
+		fi
+	done
+fi
+crypt_root_password=$(openssl passwd -crypt ${root_password})
+
+# Default shell
+clear
+gum style --border normal --margin "1" --padding "1 2" --border-foreground 212 "Select a default shell"
+shell=$(gum choose --limit 1 fish zsh bash)
+if [[ "$shell" == "fish" ]]; then
+	fish_pkg="\"fish\""
+else
+	fish_pkg=""
+fi
```

Comment:
what is this logic for? You don't have to manually install any shell, jade will handle the installation of the selected shell by itself

Reply:
when I tried fish it was not automatically installed by jade

Reply:
Just checked the jade code, it should be installing it automatically but due to a weird implementation this wouldn't always properly work so you can remove this as it was a bug on jade's side

Reply:
maybe we can just leave in until the bug is fixed?

Reply:
Yeah true